scripts/getPosts.mjs

```diff
@@ -0,0 +1,36 @@
+import apiCall from "./api.mjs";
+/**
+ * This function fetches the feed from the api with all the query parameters enabled and stores the result to sessionStorage for later use and returns the result.
+ * @returns {array} Returns the array with posts fetched from the api
+ */
+export default async function getPosts(){
+  if(!window.sessionStorage.getItem("user")){
+    window.location.href = `${window.location.origin}/login.html`;
+  };
+
+  if(!window.sessionStorage.getItem("feed")){
```

Comment:
This is causing data to become stale. When you refresh the new posts do not appear. Always use the freshest data from the API over stale data in storage.

Reply:
I used this approach mainly to reduce amount of api calls out of habit from previous assignments where we've worked with apis with restrictions. I've now changed to fetching the posts on every load

 I am still using session storage in my filter function but that's just to copy the feed object by value and the session storage is cleared every time the function is called to ensue it's always using the most recent data.